Sources/Optimizely/OptimizelyClient.swift

```diff
@@ -34,6 +34,18 @@ open class OptimizelyClient: NSObject {
             atomicConfig.property = newValue
         }
     }
+    
+    private var atomicUserContext = AtomicProperty<OptimizelyUserContext>()
+    var userContext: OptimizelyUserContext? {
```

Comment:
I'm confused.  There is a standalone user context and a user context in the optimizely client?  Where is this ever set?

Reply:
As a suggestion, can you serialize the user context to a dictionary and store that here so that you avoid holding a user context that holds a client (even weak)? It might be useful it would now just serialize client.

 Or as a less sophisticated approach you could just hold default user id and attributes and keep creating user contexts.

Reply:
Keeping a user-context was not an intention. It was a left-over while we change the architecture to createUserContext. It's cleaned up.